cpp/benchmarks/convert.cc

```diff
@@ -0,0 +1,77 @@
+//  Copyright 2022 Lance Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include <arrow/dataset/discovery.h>
+#include <arrow/dataset/file_parquet.h>
+#include <arrow/filesystem/filesystem.h>
+#include <parquet/arrow/reader.h>
+
+#include <argparse/argparse.hpp>
+#include <filesystem>
+#include <iostream>
+#include <map>
+
+#include "lance/arrow/reader.h"
+#include "lance/arrow/writer.h"
+
+using std::map;
+using std::string;
+
+namespace fs = std::filesystem;
+
+arrow::Status Convert(const std::string& in_uri, const std::string& out) {
```

Comment:
Quick docstring to note that this is Converting from parquet to lance?

Reply:
Done